tests/tests.go

```diff
@@ -50,7 +51,35 @@ func CreateTempFile(t *testing.T, content string) (*os.File, error) {
 	return file, nil
 }
 
+// CreateTempDir creates a temporary directory for testing purposes.
+// It returns the path to the created directory and an error if any.
+func CreateTempDir(t *testing.T, dirname string) (string, error) {
+	t.Helper()
+
+	// Create a temporary directory
+	dir, err := os.MkdirTemp(t.TempDir(), dirname)
+	if err != nil {
+		return "", err
+	}
+
+	// Ensure the directory is cleaned up after the test
+	t.Cleanup(func() {
+		if err := os.RemoveAll(dir); err != nil {
+			t.Fatalf("Failed to clean up temporary directory: %s", err)
+		}
+	})
+
+	return dir, nil
+}
```

Comment:
```suggestion
// CreateTempDir creates a temporary directory for testing purposes.
// It returns the path to the created directory and fail test on error.
func CreateTempDir(t *testing.T, dirname string) string {
	t.Helper()

	// Create a temporary directory
	dir, err := os.MkdirTemp(t.TempDir(), dirname)
  require.NoError (t, err)
	return dir
}
```

You have a test helper, you can fail on error, no need to return it

Also, t.TempDir already clean the created folder, so it's sub-directories too

Reply:
But then,I'm unsure why you need this helper, when you can use the.TempDir directly

Reply:
Anyway, you should consider applying this pattern to CreateTempFile

Reply:
Since I also didn't see a need for the helper to remain, I've removed it.